pkg/compose/create.go

```diff
@@ -1088,87 +1093,128 @@ func (s *composeService) ensureNetwork(ctx context.Context, n types.NetworkConfi
 	if err != nil {
 		return err
 	}
-	networkNotFound := true
+
+	// NetworkList API doesn't return the exact name match, so we can retrieve more than one network with a request
+	networks = utils.Filter(networks, func(net moby.NetworkResource) bool {
+		return net.Name == n.Name
+	})
+
 	for _, net := range networks {
-		if net.Name == n.Name {
-			networkNotFound = false
-			break
-		}
-	}
-	if networkNotFound {
-		if n.External.External {
-			if n.Driver == "overlay" {
-				// Swarm nodes do not register overlay networks that were
-				// created on a different node unless they're in use.
-				// Here we assume `driver` is relevant for a network we don't manage
-				// which is a non-sense, but this is our legacy ¯\(ツ)/¯
-				// networkAttach will later fail anyway if network actually doesn't exists
-				enabled, err := s.isSWarmEnabled(ctx)
-				if err != nil {
-					return err
-				}
-				if enabled {
-					return nil
-				}
-			}
-			return fmt.Errorf("network %s declared as external, but could not be found", n.Name)
-		}
-		var ipam *network.IPAM
-		if n.Ipam.Config != nil {
-			var config []network.IPAMConfig
-			for _, pool := range n.Ipam.Config {
-				config = append(config, network.IPAMConfig{
-					Subnet:     pool.Subnet,
-					IPRange:    pool.IPRange,
-					Gateway:    pool.Gateway,
-					AuxAddress: pool.AuxiliaryAddresses,
-				})
-			}
-			ipam = &network.IPAM{
-				Driver: n.Ipam.Driver,
-				Config: config,
-			}
+		if net.Labels[api.ProjectLabel] == n.Labels[api.ProjectLabel] &&
```

Comment:
we could invoke NetworkList with a projectFilter and networkFilter (same as `down`)  but not doing so allows to catch this scenario were a network with same name exists but doesn't have label, and `CheckDuplicate: true` will prevent us to create another one.